src/components/dialog/dialog.tsx

```diff
@@ -763,65 +764,74 @@ export class ChDialog {
                 </h2>
               )}
               <button
+                id="close-button"
                 aria-label={this.closeButtonAccessibleName || null}
                 class="close-button"
                 part="close-button"
                 type="button"
                 onClick={this.#closeHandler}
               ></button>
-            </div>
+            </header>
           )}
 
-          <div key="content" class="content" part="content">
+          <div id="content" key="content" class="content" part="content">
             <slot />
           </div>
 
           {this.showFooter && (
-            <div key="footer" class="footer" part="footer">
+            <footer key="footer" class="footer" part="footer">
```

Comment:
Same comment for the footer.

```suggestion
            <div key="footer" class="footer" part="footer">
```